app/src/main/java/com/shevchenkovtwo/homework/ui/adapters/MoviesAdapter.kt

```diff
@@ -35,22 +39,20 @@ class MoviesAdapter(private var movies: List<Movie>) :
 
     override fun onBindViewHolder(holder: MovieViewHolder, position: Int) {
         holder.bind(movies[position])
+        val bundle = bundleOf(selectedMovie to movies[position])
         holder.itemView.setOnClickListener {
-            selectedMovie = movies[position]
-            it.findNavController().navigate(R.id.action_fragmentMoviesList_to_movieDetailFragment)
+            it.findNavController()
+                .navigate(R.id.action_fragmentMoviesList_to_movieDetailFragment, bundle)
         }
     }
 
     override fun getItemCount(): Int {
         return movies.size
     }
 
-    companion object Constants {
-        var selectedMovie: Movie? = null
```

Comment:
@emartynov 
That is not the best solution. The static makes a lot of coupling and caching issues.

Reply:
reworked